### PR TITLE
chore(data-explorer): initial combination of equal logic

### PR DESCRIPTION
**Pull request by jelmerveen**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] Conventional commits (squash if needed)
- [ ] No warnings during install
- [ ] Updated javascript typing

### CODE REVIEW

File: packages/data-explorer/src/store/actions.ts [hunk 1]
```diff
@@ -26,45 +26,52 @@ export default {
...
+    if (state.dataDisplayLayout === 'CardView') {
+      dispatch('fetchCardViewData')
+    } else {
+      dispatch('fetchTableViewData')
+    }
+  },
+  /**
+   * Don't call this function directly (except for tests), use fetchViewData instead
+   */
+  fetchCardViewData: async ({ commit, state, getters }) => {
```
Comment (connoratrug) [inline packages/data-explorer/src/store/actions.ts:50]: the do not export it

File: packages/data-explorer/src/store/actions.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import * as metaDataRepository from '@/repository/metaDataRepository'
 import * as dataRepository from '@/repository/dataRepository'
 import * as metaDataService from '@/repository/metaDataService'
 import * as metaFilterMapper from '@/mappers/metaFilterMapper'
-import router from '@/router'
+import { Store } from 'vuex'
```
Comment (connoratrug) [inline packages/data-explorer/src/store/actions.ts:7]: store is never used ..

File: packages/data-explorer/src/store/actions.ts [hunk 3]
```diff
@@ -26,45 +26,52 @@ export default {
...
+    }
+  },
+  /**
+   * Don't call this function directly (except for tests), use fetchViewData instead
+   */
+  fetchCardViewData: async ({ commit, state, getters }) => {
+    const { tableName, tableMeta, tablePagination, sort, tableSettings, dataDisplayLayout } = state
+
+    // failsafe measures.
+    if (!tableName) return
```
Comment (connoratrug) [inline packages/data-explorer/src/store/actions.ts:54]: this is an error state , why would you return null

File: packages/data-explorer/src/store/actions.ts [hunk 4]
```diff
@@ -26,45 +26,52 @@ export default {
...
+      dispatch('fetchCardViewData')
+    } else {
+      dispatch('fetchTableViewData')
+    }
+  },
+  /**
+   * Don't call this function directly (except for tests), use fetchViewData instead
+   */
+  fetchCardViewData: async ({ commit, state, getters }) => {
+    const { tableName, tableMeta, tablePagination, sort, tableSettings, dataDisplayLayout } = state
```
Comment (connoratrug) [inline packages/data-explorer/src/store/actions.ts:51]: destructuring leads to vars being declared to soon, out side of context and possibly unused

**Review by connoratrug — CHANGES_REQUESTED** [review]
What is the PR trying to achieve ? What problem is it solving, and what problems is it introducing

File: packages/data-explorer/src/store/actions.ts (hunk 4 above)
Comment (jelmerveen) [reply packages/data-explorer/src/store/actions.ts:51]: Destructuring gives us more clarity upfront and removes unnecessary trailing paths, it's aready happening in the function.
unnecessary vars isn't an issue, as you only export what you need. And if there would be a rogue variable, the IDE will tell you so.

**Comment by jelmerveen** [comment]
This PR was about streamlining the dataflow through fetchViewData, as all the views are using this.
BUT there were remnants to the other ones inside vue files that made it look like the functions are all called seperately. Which was not the case and caused confusion when fixing a bug. We decided to make a seperate pull request to tackle this issue, by using the fetchViewData at all times and rely on the vuex state on the correct path. Which makes it cleaner, more readable and less prone to bugs.

Examples are: packages/data-explorer/src/views/ClipboardView.vue  
  ...mapActions(['fetchTableViewData']), is imported but never used. Suggesting it should be called directly.

And again in packages/data-explorer/src/views/TableView.vue

**Pull request closed without merging**